### ISSUE TITLE
In PhET ZIMs the load button is hidden under the bottom bar

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#435]
The title is self-explanatory, and I include a screenshot to show the problem. Note that this occurs in desktop-wide views. The button can be accessed by narrowing the window. This may be related to #386,  but as it occurs in SW mode, it may need a different solution. Or maybe we can find a generic one that solves both issues.

![image](https://user-images.githubusercontent.com/4304337/48319577-9c863200-e607-11e8-802d-8477187d4af3.png)

This may be better addressed through #120 or #337.

**Comment by kelson42** [comment]
For me it does not work at all with a recent version, even with Chrome & Service Worker.

**Comment by kelson42** [comment]
@mossroy @Jaifroid Should we really keep that ticket considering #339?

**Comment by Jaifroid** [comment]
> For me it does not work at all with a recent version, even with Chrome & Service Worker.

@kelson42, PhET ZIMs work in Service Worker mode, for which you need to run the extension version of the app (Firefox or Chrome) and select "Service Worker" in Configuration. PhET experiments work in jQuery mode only in my JavaScript Support branch, but then you have to search for the experiment title in the search bar, because the ReactJS interface is not working in that mode yet. It works fine in SW mode.

**Comment by Jaifroid** [comment]
> @mossroy @Jaifroid Should we really keep that ticket considering #339?

I think we do need to keep this ticket, because we do have PhET working in SW mode (in master). However, this ticket may be superseded by #523. Until we've made more progress on that, I think we should keep this issue.

**Comment by mossroy** [comment]
I agree with you @Jaifroid.
@kelson42 : I confirm PhET is fully working in SW mode with Firefox or Chrome, since version 2.5. Including as a Chrome extension.

**Comment by kelson42** [comment]
FYI The phets scraper has been rewritten an a version 2.0 has been published a few days ago. New ZIM files have been released. Phets related bugs should probably be re-confirmed.

**Comment by Jaifroid** [comment]
Indeed, and actually we've rewritten the algorithm for calculating the visible area, so it *may* be fixed our end. Will check.

**Comment by kelson42** [comment]
@Jaifroid Might be time to refresh that one

**Comment by Jaifroid** [comment]
I'm afraid this is still an issue in the latest PhET ZIMs I can find which are from 2021-08. Several controls are blocked by the bottom bar. It's also an issue in KJSWL, which uses the redesigned interface I aim to work on here in #523. We need to change the way the bottom bar works in relation to the iframe. It should appear below the iframe rather than using up space within it., or possibly we need to resize the iframe area so as not to incude the bottom bar pixels. This probably needs to be solved in #337 before implementing #523 because #523 doesn't fundamentally change the positioning of the bottom bar, it just allows it to be hidden on scroll like Kiwix Android.

A slightly clunky quick-fix solution might be to have a UI option that allows hiding the bottom bar, the top bar, or both, but then we'd need a way to get back to Config to unhide them when needed.

**Comment by Rbcoder1** [comment]
Can I work on this issue.

**Comment by Rbcoder1** [comment]
I Am Under This Issue [](url)

![error in phet zim](https://user-images.githubusercontent.com/107172418/229703766-cca0e4d3-072b-48ea-8d9d-c72fd051e993.png)

We Have a Problem with The Bottom Bar That Hide All Content under this

**Comment by Rbcoder1** [comment]
**I Have Solution Is that We Just Add Are Little button or icon to hide or show the Bottom Bar Component** 

OR 

**When The the Content is Load We can hide the bottom bar or show to button to display it.**

**Comment by Jaifroid** [comment]
> Can I work on this issue.

Thank you for the offer. However, we need to decide what we actually want. I don't think an extra button in the UI would be a good idea especially as we need the app to work on narrow screens, and space is already at a premium. I think it would be much better to work on #523 before trying to fix this issue. An option in configuration to hide the bottom bar, the top bar, or both might be a possibility.

**Comment by kelson42** [comment]
@Jaifroid Considering this ticket is pretty old, is pretty annoying and seems to me pretty easy to fix. Here would be my simple recommendation. As far as I can judge, having the controls in an overlay.. over the content is problematic (is #337 trigger) and I don't know other softwares doing this. I see two kinds of solution:
* Either we really want to keep a bottom bar and then it should be drawed beside the content (maybe autohidding)
* Or more usual, these control should be somewhere in the/a top part of the chrome.

@Jaifroid I would recommend to make soon a decision on this so we can merge/reorganise the few tickets talking of the same topic and focus on the implementation details.

**Comment by Rbcoder1** [comment]
> > Can I work on this issue.
> 
> Thank you for the offer. However, we need to decide what we actually want. I don't think an extra button in the UI would be a good idea especially as we need the app to work on narrow screens, and space is already at a premium. I think it would be much better to work on #523 before trying to fix this issue. An option in configuration to hide the bottom bar, the top bar, or both might be a possibility.


SoWe Can Also Fix It By Another Way 

When we scroll down At that time we will hide the bottom bar and on scroll top we will sho them again 

Or 
I try another way

**Comment by Jaifroid** [comment]
> SoWe Can Also Fix It By Another Way
> 
> When we scroll down At that time we will hide the bottom bar and on scroll top we will sho them again
> 
> Or I try another way

@Rbcoder1 What you are describing is #523. Please read that issue, try out the implementation at https://pwa.kiwix.org, and decide if you can undertake that issue. Please reply on that issue. Please note that fixing #523 probably won't fix this issue because the PhET page where the controls are hidden is not scrollable.

**Comment by Jaifroid** [comment]
Finally solved with #1142 -- after all these years.....
